src/native/host_resolver.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <jni.h>
+
+#include <aws/io/host_resolver.h>
+
+#include "crt.h"
+
+JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_io_HostResolver_hostResolverNew(JNIEnv *env,
+                                                                                        jclass jni_class,
+                                                                                        jlong jni_elg) {
+
+    struct aws_allocator *allocator = aws_jni_get_allocator();
+    struct aws_event_loop_group *el_group = (struct aws_event_loop_group *)jni_elg;
+
+    if (!el_group) {
+        aws_jni_throw_runtime_exception(env, "HostResolver.hostResolverNew: Invalid EventLoopGroup");
+        return (jlong)NULL;
+    }
+
+    struct aws_host_resolver *resolver = aws_mem_acquire(allocator, sizeof(struct aws_host_resolver));
+    aws_host_resolver_init_default(resolver, allocator, 8, el_group);
```

Comment:
This 8 should be a parameter, it's important that it's exposed, because it limits the number of hosts.

Reply:
Fixed